### DISCUSSION TITLE
mg_mgr_poll no longer accounts for timers

### DISCUSSION THREAD

**Discussion by aleiby** [cesanta/mongoose#2850] [Ask questions, get answers] [answered]
I recently finished updating from Mongoose 6.x to 7.14, and I've noticed `mg_mgr_poll` no longer accounts for timers.  There used to be a block of code with the following comment:
```
  /*
   * If there is a timer to be fired earlier than the requested timeout,
   * adjust the timeout.
   */
```
I assume this was intentional, but wanted to verify.  We've had to adjust the timeout we were using with `mg_mgr_poll` to compensate.

**Comment by scaprile** [comment, marked as answer]
https://mongoose.ws/documentation/#mg_timer_add
> NOTE: Make sure that the timer interval is equal or more to the mg_mgr_poll() timeout

https://mongoose.ws/documentation/tutorials/core/timers/
> It is important to keep the event manager polling timeout smaller than the timer interval, because timers are checked at every polling cycle. If there are no network events, the poll function would sleep for the whole polling interval and if it is larger than the timer timeout, a timer invocation could be missed.
